tests/conftest.py

```diff
@@ -1,62 +1,119 @@
-import pytest
 import os
+import sys
+import stat
 import shutil
 import subprocess
+import contextlib
+
+import pytest
 
 import imageio as iio
 
 
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "needs_internet: Marks a test that requires an active interent connection."
+        " (deselect with '-m \"not needs_internet\"').",
+    )
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--imageio-binaries",
+        action="store",
+        metavar="GIT_REPO",
+        default="https://github.com/imageio/imageio-binaries.git",
+        help="Repository to use for checking out binary data used at tests"
+        " (default: %(default)s).  Use 'file:///path/to/imageio-binaries'"
+        " to clone from a local repository and save testing bandwidth.",
+    )
+
+
+@contextlib.contextmanager
+def working_directory(path):
+    """
+    A context manager which changes the working directory to the given
+    path, and then changes it back to its previous value on exit.
+    Usage:
+    > # Do something in original directory
+    > with working_directory('/my/new/path'):
+    >     # Do something in new directory
+    > # Back to old directory
+    """
+
+    prev_cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(prev_cwd)
+
+
 @pytest.fixture(scope="session")
-def tmp_dir(tmp_path_factory):
-    # A temporary directory loaded with the test image files
-    # downloaded once in the beginning
-    tmp_path = tmp_path_factory.getbasetemp() / "image_cache"
-    tmp_path.mkdir()
-
-    # download the (only) test images via git's sparse-checkout
-    current_path = os.getcwd()
-    os.chdir(tmp_path)
-    git_version = subprocess.run(
-        ["git", "--version"], stdout=subprocess.PIPE
-    ).stdout.decode("utf-8")[12:18]
-    major, minor, _ = git_version.split(".")
-
-    if int(major) == 2 and int(minor) < 30:
-        # --sparse was introduced in git 2.30.0 (I think)
-        os.system(
-            "git clone --depth 1 --filter=blob:none --no-checkout https://github.com/imageio/imageio-binaries.git ."
-        )
-        os.system("git sparse-checkout init --cone")
-        os.system("git sparse-checkout set test-images")
-    else:
-        # sparse checkout on GH Actions
-        os.system(
-            "git clone --sparse --filter=blob:none https://github.com/imageio/imageio-binaries.git ."
-        )
-        os.system("git sparse-checkout init --cone")
-        os.system("git sparse-checkout set test-images")
-    os.chdir(current_path)
+def image_cache(pytestconfig, tmp_path_factory):
```

Comment:
Sure, please go ahead and push these to the branch.

Reply:
perfect. We did another change to that fixture in #736 which is causing the current merge conflict. 

@anjos To avoid bumping into each other, let me know when you are done with making changes here. I will then go and resolve the merge conflict and apply the changes discussed here.

Reply:
Please go ahead.  I think I'm done for this round.